### ISSUE TITLE
A *more* interactive example

### ISSUE THREAD

**Issue by alensiljak** [paulgb/interactive_process#1]
Hi! 

First of all, thanks a lot for this library. I was looking for something like `rexpect`, since it does not work on Windows.
That means I'm looking for a "conversation" with a CLI program in the sense:

send command
receive output
send another command
receive output

At the moment, the example(s) only demonstrate one handler for all output. This would result in some convoluted code for the case above, with all the output going to the same handler.
Is it possible, and what would be the simplest way to run this scenario, in your opinion?

**Comment by paulgb** [comment]
I think this would be a good use for [mpsc channels](https://doc.rust-lang.org/std/sync/mpsc/index.html). You could construct an `InteractiveProcess` with a callback that sent a message into the channel, and wrap the `send` method with a method that sends the request and then waits to receive a response from the channel.

**Comment by alensiljak** [comment]
I was thinking something similar but got bogged down with a multi-line response. This can take arbitrary time and is therefore difficult to realize when the response to a command is over.

**Comment by paulgb** [comment]
Yes, if responses can be multiple lines it gets a bit more complicated, and depends a lot on e.g. how the program indicates that it is done responding.
